ecl/hthor/hthor.cpp

```diff
@@ -958,11 +955,20 @@ void CHThorXmlWriteActivity::setFormat(IFileDescriptor * desc)
     desc->queryProperties().setProp("@format","utf8n");
     desc->queryProperties().setProp("@rowTag",rowTag.str());
     desc->queryProperties().setProp("@kind", (kind==TAKjsonwrite) ? "json" : "xml");
-    if (headerLength)
-        desc->queryProperties().setPropInt("@headerLength", headerLength);
-    if (footerLength)
-        desc->queryProperties().setPropInt("@footerLength", footerLength);
+    if (kind==TAKjsonwrite)
+    {
+        if (headerLength)
+            desc->queryProperties().setPropInt("@headerLength", headerLength);
+
+        if (footerLength)
+            desc->queryProperties().setPropInt("@footerLength", footerLength);
+    }
+    else
+    {
+        desc->queryPart(0)->queryProperties().setPropInt("@headerLength", headerLength);
+        desc->queryPart(0)->queryProperties().setPropInt("@footerLength", footerLength);
```

Comment:
it may make sense, but could do with a comment here to explain why json does not set @headerLength/@footerLength if 0, but xml does.